docs-md/developer-guide/joins/partition-data.md

```diff
@@ -98,26 +101,35 @@ When your inputs are co-partitioned, records with the same key, from
 both sides of the join, are delivered to the same stream task during
 processing.
 
-### Records Have the Same Keying Scheme
+### Records Have the Same Keying Schema
 
-For a join to work, the keys from both sides must have the same serialized
-binary data.
+For a join to work, the keys from both sides must have the same SQL type.
 
 For example, you can join a stream of user clicks that's keyed on a `VARCHAR`
 user id with a table of user profiles that's also keyed on a `VARCHAR` user id.
 Records with the exact same user id on both sides will be joined.
 
-ksqlDB requires that keys are UTF-8 encoded strings.
+If the schema of the columns you wish to join on don't match, it may be possible
+to `CAST` one side to match the other. For example, if one side of the join
+had a `INT` userId column, and the other a `LONG`, then you may choose to cast
+the `INT` side to a `LONG`:
+
+```sql
+    -- stream with INT userId
+    CREATE STREAM clicks (userId INT, url STRING) WITH(kafka_topic='clickstream', value_format='json');
+
+    -- table with BIGINT userId stored in they key:
+    CREATE TABLE  users  (ROWKEY BIGINT KEY, fullName STRING) WITH(kafka_topic='users', value_format='json');
```

Comment:
I think it would be good to have the example above explicitly name `userID` as the `KEY` - I know this isn't necessary for the example, but I think it's good to have an example of a stream with a key declared using the `KEY` syntax

Reply:
I've deliberately not added the WITH KEY bit as its not required. The example only has what's needed.  Adding additional stuff if just noise IMHO and can lead to confusion.

Plus I intend to drop the whole WITH KEY thing soon.

So do you mind if we leave this as it is?